### PR TITLE
Decouple DockerBuildWorkflow with OUTPUT_IMAGE for source container b…

### PR DESCRIPTION
**Pull request by tkdchen**
…uild

* CLOUDBLD-123

OSv3InputPlugin reads environment variable OUTPUT_IMAGE set by OpenShift
to get the image tag, and then the value will be set to
DockerBuildWorkflow image argument. However, the plugins used for
building source container image do not use the value of image from
either property workflow.image or workflow.builder.image.

To make the interface of DockerBuildWorkflow clear for both binary and
source container image build, this patch does following things to
decouple it with OUTPUT_IMAGE:

* removes image argument from DockerBuildWorkflow.
* DockerBuildWorkflow object still has a property named image, because
  plugins for binary image build also have the code 'workflow.image'.
* DockerBuildWorkflow uses self.user_params['image_tag'] internally to
  initialize InsideBuilder object by passing the value to its argument
  image.
* InsideBuilder still requires image as it is used in a few places, for
  example in BuildStepPluginRunner.run and
  BuilderStateMachine.ensure_is_built.

Signed-off-by: Chenxiong Qi <cqi@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request has a link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by tkdchen** [comment]
Hi, please review if I'm on the correct direction. Thanks!

**Comment by MartinBasti** [comment]
+1 I think that direction is good

**Comment by tkdchen** [comment]
@MartinBasti @rcerven Thanks for your review. Anything else I missed to change for the refactor?

File: atomic_reactor/inner.py [hunk 1]
```diff
@@ -483,7 +485,7 @@ def build_docker_image(self):
         :return: BuildResult
         """
         exception_being_handled = False
-        self.builder = InsideBuilder(self.source, self.image)
+        self.builder = InsideBuilder(self.source, self.user_params['image_tag'])
```
Comment (rcerven) [inline atomic_reactor/inner.py:488]: why not use already that image property? 
meaning you can leave there : self.image

File: atomic_reactor/plugins/input_osv3.py [hunk 2]
```diff
@@ -136,7 +136,6 @@ def run(self):
         build_json = get_build_json()
         git_url = os.environ['SOURCE_URI']
         git_ref = os.environ.get('SOURCE_REF', None)
-        image = os.environ['OUTPUT_IMAGE']
```
Comment (rcerven) [inline atomic_reactor/plugins/input_osv3.py:139]: there is also the same in : OSv3SourceContainerInputPlugin  line 216

File: atomic_reactor/plugins/input_osv3.py [hunk 3]
```diff
@@ -169,7 +168,6 @@ def run(self):
                     'git_branch': git_branch,
                 },
             },
-            'image': image,
```
Comment (rcerven) [inline atomic_reactor/plugins/input_osv3.py:172]: there is also the same in : OSv3SourceContainerInputPlugin  line 235

**Comment by rcerven** [comment]
we should also remove 'OUTPUT_IMAGE' from both classmethods 'is_autousable' in input_osv3.py

**Comment by tkdchen** [comment]
@rcerven Updated. PTAL.

**Comment by tkdchen** [comment]
Can this be merged before the markdownlint issue is fixed?

**Comment by rcerven** [comment]
> Can this be merged before the markdownlint issue is fixed?

yeah

**Comment by MartinBasti** [comment]
> Can this be merged before the markdownlint issue is fixed?

Yes. Lets merge it and deal with markdown later.

**Pull request merged**